language_detector/main.py

```diff
@@ -1,9 +1,35 @@
 """This is the entry point of the program."""
+# coding=utf-8
 
 from .languages import LANGUAGES
 
-
 def detect_language(text, languages=LANGUAGES):
     """Returns the detected language of given text."""
-    # implement your solution here
-    pass
+    text = text.strip().split("\n\n")
+    count = {}
+    
+    for t in text:
+        count[find_lang(t, languages=LANGUAGES)] = len(t)
+        
+    return max(count, key=count.get)
+
+
+def find_lang(text, languages=LANGUAGES):
+    for ch in "-(),.":
+        text = text.replace(ch, "")
+        
+    text = text.strip().split(" ")
+    text = [el.strip() for el in text if el != ""]
+
+    wordcount = {}
+    for word in text:
+        if word not in wordcount:
```

Comment:
+1 to that.

There is also a great built-in method for dictionaries 'get'. It lets you call a specific key, but if that doesn't exist, it becomes the default value. This loop could easily become:

```python
for word in text:
    wordcount.get(word, 0) += 1
```

You can read some more [here](https://docs.python.org/2/library/stdtypes.html#dict)